### PR TITLE
Trm 21142 results container

### PR DESCRIPTION
**Pull request by xwatkins**

### CODE REVIEW

File: src/results/state/actions.js [hunk 1]
```diff
@@ -16,10 +18,12 @@ export const fetchResultsStarted = () => ({
...
 export const fetchResults = query => (dispatch) => {
-  const queryString = createQueryString(query);
+  const uniprotQueryString = createQueryString(query);
+  const queryUrl = queryString.stringify({
+    query: uniprotQueryString,
+  });
   dispatch(fetchResultsStarted());
-  // TODO we need to use something better to handle url params
-  fetchData(`${apiUrls.advanced_search_terms}/?${queryString}`)
+  fetchData(urljoin(apiUrls.advanced_search_terms, '/?', queryUrl))
```
Comment (dlrice) [inline src/results/state/actions.js]: does this want to be `apiUrls .advanced_search`?

Comment (dlrice) [reply src/results/state/actions.js]: would url-related activities be better placed in `src/utils/apiUrls.js`? Like having the following function in `apiUrls.js`

```
export const getQueryUrl = uniprotQueryString => urljoin(
  apiUrls.advanced_search_terms,
  '/?',
  queryString.stringify({ query: uniprotQueryString }),
);
```
then in `src/results/state/actions.js` it is called: `fetchData(getQueryUrl(uniprotQueryString))`.

File: src/results/state/actions.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+export const fetchResults = query => (dispatch) => {
+  const uniprotQueryString = createQueryString(query);
+  const queryUrl = queryString.stringify({
+    query: uniprotQueryString,
+  });
+  dispatch(fetchResultsStarted());
+  fetchData(urljoin(apiUrls.advanced_search_terms, '/?', queryUrl))
+    .then(response => dispatch(fetchResultsSuccess(response.data)))
+    .catch(error => throw error);
```
Comment (dlrice) [inline src/results/state/actions.js]: Will want to add `@babel/plugin-proposal-throw-expressions` to dev deps & babelrc to support the `throw` statement.

Comment (xwatkins) [reply src/results/state/actions.js]: We still need to decide on error handling in the application. Reverting to console.error for now

**Comment by xwatkins** [comment]
Changes made, have a look

**Pull request merged**